packages/core/src/agent/Agent.ts

```diff
@@ -201,10 +201,10 @@ export class Agent {
 
     // Stop transports
     for (const transport of this.outboundTransports) {
-      transport.stop()
+      await transport.stop()
     }
     for (const transport of this.inboundTransports) {
-      transport.stop()
+      await transport.stop()
     }
```

Comment:
Wouldn't be better to do something like this `await Promises.all(this.inboundTransports.map(t => t.stop()))`? I think we just need to wait until all transports finish

Reply:
Yes! I'll address in a seperate PR